src/error.ts

```diff
@@ -150,7 +150,7 @@ export class MongoError extends Error {
   }
 
   /** @internal */
-  static buildErrorMessage(e: Error | string): string {
+  static buildErrorMessage(e: unknown): string {
```

Comment:
What is the motivation for this change? I don't see where in the code it could be set differently rather than `Error` or `string`.

Reply:
I made the type in `connectionFailureError` more accurate which makes error nullish. To me it felt easier to propagate the handling of that to `buildErrorMessage`

Reply:
Isn't it then `Error | string | null`?

Reply:
Any error caught in a catch block has the potential to be any type, we just disabled that rule temporarily, `useUnknownInCatchVariables`, I'm writing unknown here now so there is less work for NODE-3659 down the road.